### PR TITLE
Support for EIP-2930 and EIP-1559

### PR DESCRIPTION
**Pull request by i-norden**
This includes the updates from https://github.com/ipld/specs/pull/366, also introduces support for the type modifications in [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md), as well as miscellaneous updates/fixes discovered while working on the [codec implementations](https://github.com/vulcanize/go-codec-dageth).

### CODE REVIEW

**Comment by i-norden** [comment]
I can't seem to make sense of the errors that are being thrown by fleek, it is saying "bad link" for lines which don't contain links (in some cases the lines are just part of the markdown and not part of the ipld schema sections or type defs at all).

File: specs/codecs/dag-eth/chain.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Ethereum Chain Data Structures
+
+This section contains the IPLD schemas for the blockchain data structures of Ethereum.
+This includes: headers, uncle sets, transactions, and receipts. The state trie, storage trie,
+receipt trie, and transaction trie IPLDs are described in the [state](./state.md) section. It
```
Comment (rvagg) [inline specs/codecs/dag-eth/chain.md:5]:
```suggestion
receipt trie, and transaction trie IPLDs are described in the [state](../state) section. It
```

File: specs/codecs/dag-eth/index.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ For more information about the IPLD Schema language, see the [IPLD Schema docume
...
 ## Data Structure Descriptions
 
-* [Ethereum Data Structures **Basic Types**](./basic_types/)
-* [Ethereum **Chain** Data Structures](./chain/)
-* [Ethereum **Convenience Types**](./convenience_types/)
-* [Ethereum **State** Data Structures](./state/)
+* [Ethereum Data Structures **Basic Types**](./basic_types.md)
+* [Ethereum **Chain** Data Structures](./chain.md)
+* [Ethereum **Convenience Types**](./convenience_types.md)
+* [Ethereum **State** Data Structures](./state.md)
```
Comment (rvagg) [inline specs/codecs/dag-eth/index.md:24]:
```suggestion
* [Ethereum Data Structures **Basic Types**](./basic_types)
* [Ethereum **Chain** Data Structures](./chain)
* [Ethereum **Convenience Types**](./convenience_types)
* [Ethereum **State** Data Structures](./state)
```

File: specs/codecs/dag-eth/index.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
-title: "Specs: Ethereum"
-navTitle: "Ethereum"
+title: "Specs: DAG-ETH"
+navTitle: "DAG-ETH"
```
Comment (rvagg) [inline specs/codecs/dag-eth/index.md:3]:
```suggestion
navTitle: "DAG-ETH"
eleventyNavigation:
  synopsys: "Specification for Ethereum as an IPLD Data Structure and the suite of codecs used to convert to and from the IPLD Data Model"
```

.. or adapt wording as you think appropriate - we don't have enough of these synopsys' on the site yet and "DAG-ETH" is now less descriptive than just "Ethereum". The synopsys shows up on the index page that links to each codec spec.

File: specs/codecs/dag-eth/convenience_types.md [hunk 4]
```diff
@@ -65,7 +62,7 @@ the sets of transactions and receipts for that block in order to avoid the need
 and receipt tries to collect these sets (as is required when starting from a canonical `Header` block).
 These objects can be generated or verified by following the links within the contained `Header` to collect the `Transactions` and `Receipts`
 from the referenced transaction and receipt tries.
-* The IPLD block is a CBOR serialization of the object
+* The IPLD block is a CBOR serialization of the Block object.
```
Comment (rvagg) [inline specs/codecs/dag-eth/convenience_types.md:65]: `Block` has links in it, so this should probably be `DAG-CBOR` I think?

```suggestion
* The IPLD block is a DAG-CBOR serialization of the `Block` object.
```

File: specs/codecs/dag-eth/convenience_types.md [hunk 5]
```diff
@@ -100,37 +97,35 @@ Provided a `Header` multihash/CID, we can generate a `Block` IPLD by
...
+# Receipts contains a list of all of the receipts at this block
+type Receipts [Receipt]
 ```
 
 ## Genesis IPLD
 
 This is the IPLD schema for the configuration settings and genesis allocations to produce a specific genesis block and begin an Ethereum
 blockchain. It also includes a reference to the genesis block `Header` it produces. This is a single IPLD block at the base of an entire Ethereum chain.
-* The IPLD block is a CBOR serialization of the object
+* The IPLD block is a CBOR serialization of the GenesisInfo object.
```
Comment (rvagg) [inline specs/codecs/dag-eth/convenience_types.md:128]:
```suggestion
* The IPLD block is a CBOR serialization of the `GenesisInfo` object.
```

**Review by rvagg — CHANGES_REQUESTED** [review]
Such great info in here! I hope it gets enough of an audience to warrant the work, I suspect this is one of the most comprehensive sources of information about Ethereum block layouts from a content-addressing perspective available, so 👌.
A few minor nits/suggestions and some fixes for the link checker. It's annoying, and I'm certainly not going to defend the site builder here but it wants to turn every non-index.md doc into its own directory, so your links need to deal with that—which also means the markdown version of your links are broken because it apparently doesn't know how to rewrite links .. or .. something 🤷. But the suggestions for links in here will stop the complaints.

File: specs/codecs/dag-eth/convenience_types.md (hunk 4 above)
Comment (warpfork) [reply specs/codecs/dag-eth/convenience_types.md:65]: Somewhat confused by this actually -- are CBOR objects present in Ethereum already?  Or are CBOR objects being synthesized here?

Comment (warpfork) [reply specs/codecs/dag-eth/convenience_types.md:65]: If being synthesized -- why does their codec matter, what does that come from?  Did they need to get a CID generated for some reason?

**Comment by i-norden** [comment]
@rvagg @warpfork appologies for the slow turn around, I missed the notification on this. Resolved most the comments, left the DAG-CBOR one open for further discussion.

**Comment by rvagg** [comment]
👍 I don't have any concerns around the DAG-CBOR / synthetic block issue but it'll be interesting to see this all materialise and be used, maybe more opinions will arise out of that. But I can see this pattern of needing synthetic structures to make the actual structures more useful/usable being a common pattern in these retrofitting exercises.

**Comment by rvagg** [comment]
not intending to shut that discussion down btw, I just don't think it's worth blocking a merge on but it might be worth digging in deeper to the details and adjusting course based on feedback

**Pull request merged**